src/org/ods/component/HelmDeploymentStrategy.groovy

```diff
@@ -195,14 +196,113 @@ class HelmDeploymentStrategy extends AbstractDeploymentStrategy {
                         'helmAdditionalFlags': options.helmAdditionalFlags,
                     ])
                 rolloutData["${resourceKind}/${resourceName}"] = podData
+
+                // We need to find the pod that was created as a result of the deployment.
+                // The previous pod may still be alive when we use a rollout strategy.
+                // We can tell one from the other using their creation timestamp,
+                // being the most recent the one we are interested in.
+                def latestPods = getLatestPods(podData)
+                // While very unlikely, it may happen that there is more than one pod with the same timestamp.
+                // Note that timestamp resolution is seconds.
+                // If that happens, we are unable to know which is the correct pod.
+                // However, it doesn't matter which pod is the right one, if they all have the same images.
+                def sameImages = haveSameImages(latestPods)
+                if (!sameImages) {
+                    throw new RuntimeException(
+                        "Unable to determine the most recent Pod. " +
+                        "Multiple pods running with the same latest creation timestamp " +
+                        "and different images found for ${resourceName}"
+                    )
+                }
                 // TODO: Once the orchestration pipeline can deal with multiple replicas,
                 // update this to store multiple pod artifacts.
                 // TODO: Potential conflict if resourceName is duplicated between
                 // Deployment and DeploymentConfig resource.
-                context.addDeploymentToArtifactURIs(resourceName, podData[0]?.toMap())
+                context.addDeploymentToArtifactURIs(resourceName, latestPods[0]?.toMap())
             }
         }
-        rolloutData
+        return rolloutData
+    }
+
+    /**
+     * Returns the pods with the latest creation timestamp.
+     * Note that the resolution of this timestamp is seconds and there may be more than one pod with the same
+     * latest timestamp.
+     *
+     * @param pods the pods over which to find the latest ones.
+     * @return a list with all the pods sharing the same, latest timestamp.
+     */
+    @NonCPS
+    private static List getLatestPods(Iterable pods) {
+        return maxElements(pods) { it.podMetaDataCreationTimestamp }
     }
 
+    /**
+     * Checks whether all the given pods contain the same images, ignoring order and multiplicity.
+     *
+     * @param pods the pods to check for image equality.
+     * @return true if all the pods have the same images or false otherwise.
+     */
+    @NonCPS
+    private static boolean haveSameImages(Iterable pods) {
+        return areEqual(pods) { a, b ->
+            def imagesA = a.containers.values() as Set
```

Comment:
Would comment https://github.com/opendevstack/ods-jenkins-shared-library/blob/4836a2959053403cf8175f1639e1c15ff849de9c/src/org/ods/services/OpenShiftService.groovy#L1282 come into play here?

Reply:
Again, I would say it's out of the scope. This is sensible code and making a mistake here could compromise traceability. But it's something to take into account for a future PR.